### ISSUE TITLE
DDA monsters have multiple attacks of subtype spell, ignores all but last

### ISSUE THREAD

**Issue by Shad0wBlad321** [chaosvolt/cdda-arcana-mod#437]
DEBUG    : mon_archon specifies more than one attack of (sub)type spell, ignoring all but the last.  Add different `id`s to each attack of this type to prevent this.

 FUNCTION : void mtype::add_special_attack(const JsonObject&, const std::string&)
 FILE     : src/monstergenerator.cpp
 LINE     : 1495
 VERSION  : cdda-experimental-2024-08-04-2117 5dca8af

**Comment by JohnHenryEton** [comment]
Getting this too, prevents loading world.

**Comment by chaosvolt** [comment]
Ugh, I need to figure out how to fix it sometime, been busy a lot lately.

**Comment by JohnHenryEton** [comment]
Take your time. Just making you aware. Don't feel obligated to do it fast if you're busy since you're doing this for free after all.

**Comment by Keltas** [comment]
It seems to be simply requiring a unique ID to each attack when a monster has multiple special attacks that are of the same type (in this case spell).
By adding "id": "arcana_monster_archon_stare_1" in the example below the error goes away.
{ "type": "spell", "id": "arcana_monster_archon_stare_1", "spell_data": { "id": "stare" }, "monster_message": "", "cooldown": 50 },

They seem to have also changed the "see_cost" from an integer to a string. I don't know what the new values mean I have substituted 5 with "high", 2 with "low" and added "see_cost": "none" where there is empty space as the game now seems to require it. I have attached my version of the updated files that allow me to run the game seemingly without errors. My apologies for not doing a pull request however I am not a programmer and have no idea how to use git.

[monsters.json](https://github.com/user-attachments/files/16547980/monsters.json)
[overmap_terrain.json](https://github.com/user-attachments/files/16547992/overmap_terrain.json)
